### ISSUE TITLE
Path autocompletion is broken on application create panel

### ISSUE THREAD

**Issue by alexmt** [argoproj/argo-cd#2269]
**Describe the bug**

~~Looks like the happy path is broken:~~ Path autocompletion is broken on application create panel.

**To Reproduce**

1. Try to create any app using Argo CD UI and Git repo.
2. Make sure repo is not cloned into temp directory.
3. ~~Repo server keep complaining that repo does not have app~~ See path field has no suggestions

**Comment by alexmt** [comment]
Updated ticket description. Only path autocompletion is broken. Because I've used incorrect directory name since autocompletion is broken, so I've decided it is also impossible to create an app. 

Autocompletion does not work because `err := g.client.Checkout(resolvedRevision)` is missing in `ListApps` method:

https://github.com/argoproj/argo-cd/blob/4e9772e19b8356e55a9cfe8f2ba5e4398e650f5b/util/git/repo/repo.go#L40
